scripts/inner/kubestellar

```diff
@@ -75,45 +74,40 @@ wait_for_process(){
   echo $status_code
 }
 
-
 function ensure_imws() {
-    initial_ws=$(kubectl ws . --short)
     while IFS=',' read -ra items; do
         for i in "${items[@]}"; do
             if [ "$i" == "" ] ; then
                 continue
             fi
+            in_cluster=
+            if $IN_CLUSTER; then
+                in_cluster="--in-cluster"
+            fi 
+
             imw=${i#"root:"}
             echo "Ensuring IMW: $imw"
-            kubectl ws root &> /dev/null
-            if ! kubectl ws "$imw" &> /dev/null ; then #
-                if ! kubectl ws create "$imw" --enter &> /dev/null ; then
-                    echoerr 'unable to create "$imw"!'
-                    break
-                fi
-            fi
-            kubestellar-kube-bind "$imw" "synctargets" --start-konnector true
-            kubestellar-kube-bind "$imw" "locations"
+            space_config="${KUBECONFIG_DIR}/${PROVIDER_NAMESPACE}-${imw}"
+            kubectl-kubestellar-create-space --space-name $imw --provider-name $PROVIDER_NAME --sm-core-config $SPACE_MANAGER_KUBECONFIG
```

Comment:
So the name of this function should be changed to `create_imws`, right?
What if the init container crashes and gets restarted?

Reply:
Never mind, I later discovered that `kubectl-kubestellar-create-space` is named wrongly.